### PR TITLE
Modules to build kinesis

### PR DESCRIPTION
**Pull request by crissupb**
Modules to build kinesis datastreams and firehose.

Functional WIP

### CODE REVIEW

File: firehose_to_redshift/main.tf [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        compression_format = "GZIP"
+        kms_key_arn = "${var.s3_temp_key_arn}"
+        prefix = "${var.s3_intermediate_bucket_prefix}"
+    }
+
+    redshift_configuration {
+        role_arn = "${aws_iam_role.firehose_to_redshift.arn}"
+        cluster_jdbcurl = "${var.redshift_jdbc}"
+        username = "${var.redshift_username}"
+        password = "${var.redshift_password}"
```
Comment (brodygov) [inline firehose_to_redshift/main.tf]: Same comment from other PR about not putting secrets in Terraform.

File: lambda/main.tf [hunk 2]
```diff
@@ -0,0 +1,64 @@
+data "aws_caller_identity" "current" {}
```
Comment (brodygov) [inline lambda/main.tf]: Would be nice to include a short README in the `lamba/` folder explaining the purpose of the module. Looks like the primary benefit is that it automatically creates the right role for your lambda to be able to run and write logs, right?

File: s3bucket/main.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_s3_bucket" "bucket" {
+    bucket = "login-gov-${var.bucket_name}-${var.env_name}-${data.aws_caller_identity.current.account_id}-${var.region}"
+    acl    = "private"
```
Comment (brodygov) [inline s3bucket/main.tf]: I think this should probably include an empty `policy { }` section. Based on experience with the prod logs bucket it seems that Terraform won't revert manual changes to the bucket policy unless the section is explicitly set.

File: secret_with_rotation/main.tf [hunk 4]
```diff
@@ -0,0 +1,15 @@
+resource "aws_secretsmanager_secret" "secret_with_rotation" {
```
Comment (brodygov) [inline secret_with_rotation/main.tf]: Is this module used anywhere yet? Interested to see how it works out.

File: s3bucket_temporary/main.tf [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    lifecycle_rule {
+        id      = "lifecycle"
+        enabled = true
+
+        expiration {
+            days = 180
+        }
+    }
+
```
Comment (brodygov) [inline s3bucket_temporary/main.tf]: Same note about adding an empty `policy { }` here.

File: firehose_to_redshift/main.tf (hunk 1 above)
Comment (crissupb) [reply firehose_to_redshift/main.tf]: See notes in identity-devops about secrets for this service.

File: secret_with_rotation/main.tf (hunk 4 above)
Comment (crissupb) [reply secret_with_rotation/main.tf]: It is being used now in terraform-analytics-common

File: s3bucket/main.tf (hunk 3 above)
Comment (crissupb) [reply s3bucket/main.tf]: If I try to set an empty policy I get this error "policy must be a single value, not a list".

**Pull request closed without merging**